Fixes incorrect integration id

gurumitts opened the pull request:
## Description:

This PR will fix issues users with multiple Lutron Caseta devices were having.  The root cause of the issue was a difference in IDs being reported by the Lutron Integration report and the values received by the SSH /devices call.  This has been resolved by fixing the underlying library to only use SSH.  Only minor changes were required to this code base.  

**Related issue (if applicable):** fixes #6870

**Pull request in [home-assistant.github.io](https://github.com/home-assistant/home-assistant.github.io) with documentation (if applicable):** home-assistant/home-assistant.github.io#2453
## Example entry for `configuration.yaml` (if applicable):
```yaml
lutron_caseta:
    host: 192.168.XX.XXX
```

## Checklist:

If user exposed functionality or configuration variables are added/changed:
  - [x] Documentation added/updated in [home-assistant.github.io](https://github.com/home-assistant/home-assistant.github.io)

If the code communicates with devices, web services, or third-party tools:
  - [x] Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**
  - [ ] New dependencies have been added to the `REQUIREMENTS` variable ([example][ex-requir]).
  - [ ] New dependencies are only imported inside functions that use them ([example][ex-import]).
  - [x] New dependencies have been added to `requirements_all.txt` by running `script/gen_requirements_all.py`.
  - [ ] New files were added to `.coveragerc`.

If the code does not interact with devices:
  - [x] Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**
  - [ ] Tests have been added to verify that the new code works.

[ex-requir]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L14
[ex-import]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L54

gurumitts committed:
Fixes incorrect integration id

gurumitts committed:
Fixes incorrect integration id

gurumitts committed:
fixes requrements

gurumitts committed:
fixes requrements

gurumitts commented:
That's weird i run with 3.5:
```
___________________________________ summary ____________________________________
SKIPPED:  py34: InterpreterNotFound: python3.4
  py35: commands succeeded
SKIPPED:  py36: InterpreterNotFound: python3.6
  lint: commands succeeded
  requirements: commands succeeded
```
I will look into this more tomorrow.

robbiet480 committed:
Fix iOS icon calculation to return mdi:battery for levels above 95

robbiet480 committed:
Change iOS sensor unique ID to use the device ID

robbiet480 committed:
Track device last identify time

fabaff commented on homeassistant/components/lutron_caseta.py:
```diff
@@ -41,9 +41,7 @@ def setup(hass, base_config):
     hass.data[LUTRON_CASETA_SMARTBRIDGE] = Smartbridge(
-        hostname=config[CONF_HOST],
-        username=config[CONF_USERNAME],
-        password=config[CONF_PASSWORD]
```
Optional username and password suggest that users still can provide them if the setup requires it. This change looks like that authentication with username/password is no longer supported.

If so, this would become a breaking change.

gurumitts replied:
The user name and password are no longer required.  The lutron ssh integration works with a hardcoded key/user that cannot be changed.  I made them optional​ just in case some folks already had them configured.  Should I just removed them?  The allow extra should allow already configured username and password to continue working.  Basically they will be ignored.

balloob replied:
We should not allow any config options that are not used. Please remove them and let's mark this a breaking change.

Mister-Espria committed:
small fix for random effect in order to use the whole rgb range. So 255 is not excluded anymore. (#7156)

cmsimike committed:
exposed content_type in rest_command (#7101)

* exposed content_type in rest_command, which allows for manually specifying the content_type for more-strict api endpoints

* fixed up column length

Length was 86 chars, and it needed to be 79

* double import of HTTP_HEADER_CONTENT_TYPE

Removed the accidental double-import of HTTP_HEADER_CONTENT_TYPE

* moved rest_command-specific config value into component

* if no content_type, default to None

* unit test

* newline

* unused CONTENT_TYPE_TEXT_PLAIN

* removed the http-agnostic abstraction hass provided in favor of aiohttps hdrs constant

pavoni committed:
Add vera power meter. (#7134)

* Add vera power meter.

* Use W for power.

gurumitts committed:
Merge branch 'fix_integrtion_ids' of github.com:gurumitts/home-assistant into fix_integrtion_ids

balloob commented:
It seems like you have some commits in this branch that are not yours, please clean it up.

gurumitts commented:
@Balloob yep... Working on it.  I am also having issues with the tests.  I may send a new PR. I will reference this one in the new one if that is the route I take.  Sorry about this

gurumitts commented:
Closing this one in favor of https://github.com/home-assistant/home-assistant/pull/7165
